shared/src/main/scala/amf/plugins/document/vocabularies/emitters/instances/DeclarationsGroupEmitter.scala

```diff
@@ -0,0 +1,108 @@
+package amf.plugins.document.vocabularies.emitters.instances
+import amf.core.annotations.LexicalInformation
+import amf.core.emitter.{EntryEmitter, RenderOptions, SpecOrdering}
+import amf.core.parser.Position
+import amf.core.parser.Position.ZERO
+import amf.plugins.document.vocabularies.model.document.{Dialect, DialectInstanceUnit}
+import amf.plugins.document.vocabularies.model.domain.{DialectDomainElement, NodeMappable, PublicNodeMapping, UnionNodeMapping}
+import org.yaml.model.YDocument.EntryBuilder
+import org.yaml.model.YNode
+import amf.core.utils.AmfStrings
+
+case class DeclarationsGroupEmitter(declared: Seq[DialectDomainElement],
+                                    publicNodeMapping: PublicNodeMapping,
+                                    nodeMappable: NodeMappable,
+                                    instance: DialectInstanceUnit,
+                                    dialect: Dialect,
+                                    ordering: SpecOrdering,
+                                    declarationsPath: Seq[String],
+                                    aliases: Map[String, (String, String)],
+                                    keyPropertyId: Option[String] = None,
+                                    renderOptions: RenderOptions)
+    extends EntryEmitter
+    with DialectEmitterHelper {
+
+  def computeIdentifier(decl: DialectDomainElement): String = {
+    decl.declarationName.option() match {
+      case Some(name) => name
```

Comment:
Field that semantically represents name? Like schema-org#name? or http://a.ml/vocabularies/amf/core##Name, shacl#Name (in a first version maybe anything that has a name "Name" : .#Name?)

Reply:
Good suggestion, keeping it in mind for further refactors.